File: FinanceApp/FinanceApp/ClientApp/src/app/models/CashFlowItem.ts
import {CurrencyType} from "./Currency";
import {RecurrencyPeriodType} from "./RecurrencyPeriod";

export class CashFlowItem {
  constructor(id: string, name: string, otherParty: string, type: CashFlowDirection, amount: number, currency: CurrencyType, recurrencyPeriod: RecurrencyPeriodType) {
    this.id = id;
    this.name = name;
    this.otherParty = otherParty;
    this.type = type;
    this.amount = amount;
    this.currency = currency;
    this.recurrencyPeriod = recurrencyPeriod;
  }

  id: string;
  name: string;
  otherParty: string;
  type: CashFlowDirection;
  amount: number;
  currency: CurrencyType;
  recurrencyPeriod: RecurrencyPeriodType;
}

export enum CashFlowDirection {
  Income = 'INCOME',
  Cost = 'COST'
}

File: FinanceApp/FinanceApp/Models/CashFlowItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using FinanceApp.Enums;

namespace FinanceApp.Models;

public class CashFlowItem
{
    [Required] [Key] public int Id { get; set; }

    [Required] public string Name { get; set; }

    [Required] public string OtherParty { get; set; }

    [Required] public int Amount { get; set; }

    [Required] public Currency Currency { get; set; }

    [Required] public CashFlowDirectionType Type { get; set; }
    
    [Required] public RecurrencyPeriod RecurrencyPeriod { get; set; }

    [JsonIgnore] public ApplicationUser? User { get; set; }
    [JsonIgnore] [ForeignKey("User")] public string? UserId { get; set; }
}